packages/core-data/src/actions.js

```diff
@@ -256,13 +256,16 @@ export function* saveEntityRecord(
 			// to the actual persisted entity if the edits don't
 			// have a value.
 			let data = { ...persistedRecord, ...autosavePost, ...record };
-			data = Object.keys( data ).reduce( ( acc, key ) => {
-				if ( [ 'title', 'excerpt', 'content' ].includes( key ) ) {
-					// Edits should be the "raw" attribute values.
-					acc[ key ] = get( data[ key ], 'raw', data[ key ] );
-				}
-				return acc;
-			}, {} );
+			data = Object.keys( data ).reduce(
+				( acc, key ) => {
+					if ( [ 'title', 'excerpt', 'content' ].includes( key ) ) {
+						// Edits should be the "raw" attribute values.
+						acc[ key ] = get( data[ key ], 'raw', data[ key ] );
+					}
+					return acc;
+				},
+				{ status: data.status === 'auto-draft' ? 'draft' : data.status }
+			);
```

Comment:
This part is very specific to the "post" autosave. I don't mind for it being in core-data right now because that's the only auto-save but in a generic way, this could be handled on the backend somehow, just send everything and let the backend do its thing.

For now maybe we can just add some comments to clarify. Also we might want to add e2e tests for the URL changes...

Reply:
There is probably a filter we could use.